src/main/java/ch/njol/skript/expressions/ExprBuildRestrictions.java

```diff
@@ -0,0 +1,148 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.bukkitutil.BukkitUnsafe;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.*;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import com.destroystokyo.paper.Namespaced;
+import org.bukkit.Material;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.*;
+
+@Name("Build Restrictions Of Item")
+@Description("Get or modify the build restrictions of an item.")
+@Examples({"add dirt to destroyable keys of player's tool",
+	"add stone and diamond ore to destroy blocks of {_item}",
+	"clear destroyable items of {_item}",
+	"remove sand from destroy keys of {_item}"})
+@Since("INSERT VERSION")
+@RequiredPlugins("Paper")
+public class ExprBuildRestrictions extends PropertyExpression<ItemType, ItemType> {
+
+	static {
+		if (Skript.methodExists(ItemMeta.class, "getDestroyableKeys")) {
+			Skript.registerExpression(ExprBuildRestrictions.class, ItemType.class, ExpressionType.PROPERTY,
+				"[the] destroy[able] (key|block|item)[s] (of|on) %itemtypes%",
+				"[the] place[able| on] (key|block|item)[s] (of|on) %itemtypes%");
```

Comment:
we call it build restrictions but never use that in the syntax? These patterns should have something like `the build restrictions of %item%` and `the break/destroy restrictions of %item%`.
I'd also force the use of `able` in the syntaxes. `the place items of {_item}` doesn't make much sense imo.

Reply:
I'm gonna resolve this as I've just commited a new version which (I think) works a bit better in line with this. Let me know what you think though, I'm still a bit iffy with the patterns.